### PR TITLE
Pre-commit hook: ignore ES6 JS files in pre-commit hook

### PR DESCRIPTION
**Pull request by brbrr**
#### Changes proposed in this Pull Request:

* Ignore non ES6 JS files in `eslint` pre-commit hook

#### Testing instructions:

* checkout the branch
* add some styling violations into any js file in modules folder. try to commit - no errors should be thrown by pre-commit hook
* add some styling violations into any js file in `_inc/client`. try to commit - eslint should block the commit. 

#### Proposed changelog entry for your changes:

* nothing 🎶

### CODE REVIEW

File: bin/pre-commit-hook.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const gitFiles = execSync( 'git diff --cached --name-only --diff-filter=ACM' )
 	.toString()
 	.split( '\n' )
 	.map( name => name.trim() );
-const jsFiles = gitFiles.filter( name => name.endsWith( '.js' ) || name.endsWith( '.jsx' ) );
+const jsFiles = gitFiles.filter( name => name.startsWith( '_inc' ) )
```
Comment (jeherve) [inline bin/pre-commit-hook.js]: That may not be enough I am afraid, as we have old js files in the `_inc` directory as well. `_inc/twitter-timeline.js` is a good example.

Maybe we should limit this to `_inc/client`?

**Review by jeherve — DISMISSED** [review]
Quick remark

File: bin/pre-commit-hook.js (hunk 1 above)
Comment (brbrr) [reply bin/pre-commit-hook.js]: That makes sense. Updated to `_inc/client` in bbe8f53

**Review by jeherve — APPROVED** [review]
this should be good now! 🚢

**Pull request merged**